pkg/service/types.go

```diff
@@ -51,33 +49,17 @@ type MeshService struct {
 	Protocol string
 }
 
-// NamespacedKey is the key (i.e. namespace + ProviderKey()) with which to lookup the backing service within the provider
+// NamespacedKey is the key (i.e. namespace + Name) with which to lookup the backing service within the provider
 func (ms MeshService) NamespacedKey() string {
-	return fmt.Sprintf("%s/%s", ms.Namespace, ms.ProviderKey())
+	return fmt.Sprintf("%s/%s", ms.Namespace, ms.Name)
 }
 
-// Subdomain is an optional subdomain for this MeshService
-// TODO: possibly memoize if performance suffers
-func (ms *MeshService) Subdomain() string {
-	nameComponents := strings.Split(ms.Name, ".")
-	if len(nameComponents) == 1 {
-		return ""
+// FullName is the name of the service including the subdomain.
+func (ms MeshService) FullName() string {
```

Comment:
This change is the sneakiest IMO; this means that subdomain is not included by default. Is there an instance where we don't want the subdomain? It feels like there are several mistakes that could result from this change. Additionally, this change prevents Name from being a compound primary key (along with port) over the set of MeshServices. Is this intended?